app/helpers/items_helper.rb

```diff
@@ -1,11 +1,8 @@
 module ItemsHelper
   def bought_by item, user
-    orders = user.orders
-    line_items = orders.map { |order| order.line_item }
-    items = line_items.map { |line_item| line_item.item }
-    if items.include? self
-      return true
-    end
-    return false
+    return false unless user
+    items = user.orders.where(status: 'completed').
+      flat_map(&:items)
+    return items.include? item
```

Comment:
It is exactly like the flatten method, just combined with the map method all in one. It is the same as calling map(something).flatten, except it is less code and has faster performance.